setuptools/msvc.py

```diff
@@ -21,9 +21,10 @@
 import itertools
 import subprocess
 import distutils.errors
+import sys
 from setuptools.extern.more_itertools import unique_everseen
 
-if platform.system() == 'Windows':
+if sys.platform == "win32":
```

Comment:
Please don't change the semantics.  If mypy can't handle `platform.system()` then disable the checks.

Reply:
I'll change back and just ignore related platform-specific type errors in this file for now. I also disagree about this non-obtuse platform check form not being checked by type-checkers.
Here's a mypy issue we can push forward: https://github.com/python/mypy/issues/8166
pyright almost certainly wouldn't include it until it's explicitly part of a pep (it wasn't in https://peps.python.org/pep-0484/#version-and-platform-checking) or mypy also supports it.